lightning/restore/restore.go

```diff
@@ -490,12 +492,17 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	}
 	taskCh := make(chan task, rc.cfg.App.IndexConcurrency)
 	defer close(taskCh)
+	oriGCLifeTime, err := ObtainGCLifeTime(ctx, rc.tidbMgr.db)
+	if err != nil {
+		return err
+	}
+	ctx2 := context.WithValue(ctx, gcLifeTimeKey, oriGCLifeTime)
```

Comment:
Err no you do need a *pointer* here. Every `struct{}` are the same but  `*struct{}` can be different.

```suggestion
	ctx2 := context.WithValue(ctx, &gcLifeTimeKey, oriGCLifeTime)
```